extensions/wikia/PageShare/scripts/PageShare.js

```diff
@@ -72,8 +62,40 @@ require(['wikia.window', 'wikia.tracker', 'jquery'], function(win, tracker, $) {
 		});
 	}
 
-	// bind events to links
-	$(function() {
-		loadShareIcons();
-	});
+	/**
+	 * @desc Returns a language code.
+	 * If user is logged in, language code is read from user's preferences.
+	 * For anonymous users first two letters of the browser/system regional preferences are used.
+	 * Both values are ignored and overwritten if function is provided with a non-false parameter.
+	 *
+	 * @param {*} useLangQueryStringParam
+	 * @returns {String|null}
+	 */
+	function getShareLang(useLangQueryStringParam) {
+		// forced by param
+		if (useLangQueryStringParam) {
+			return useLangQueryStringParam;
+		// logged in user
+		} else if (win.wgUserName) {
+			return win.wgUserLanguage;
+		// anonymous user who uses
+		// Chrome or Firefox
+		} else if (win.navigator.languages) {
+			return win.navigator.languages[0].substr(0, 2);
+		// Safari
+		} else if (win.navigator.language) {
+			return win.navigator.language.substr(0, 2);
+		// Internet Exploder
+		} else if (win.navigator.userLanguage) {
+			return win.navigator.userLanguage.substr(0, 2);
+		// something exotic
+		} else {
+			return null;
```

Comment:
Awesome idea, thanks! Shall i do it like this?
```
wikia.log(['Couldn't pick a language for Page Share extension.'], 'debug', page_share);
```
Is it sufficient? Or should I add some variables as next messages in the array? Does it even work like that?